web/src/features/fireWeather/components/maps/stationAccuracy.test.ts

```diff
@@ -1,155 +1,509 @@
 import {
-  computeAccuracyColors,
   StationMetrics,
   neutralColor,
   noDataColor,
   tempColorScale,
   rhColorScale,
-  computePercentageDifference,
-  computeScaleIndex,
   neutralIndex,
-  differenceToMagnitude
+  computeTempAccuracyColor,
+  computeTempAccuracySize,
+  computeTempScaleIndex,
+  computeRHAccuracyColor,
+  computeRHAccuracySize,
+  computeRHScaleIndex,
+  darkOrangeColor,
+  darkGreenColor,
+  lightGreenColor,
+  middleOrangeColor,
+  darkRedColor,
+  darkBlueColor,
+  pinkColor,
+  mediumBlueColor,
+  smallRadius,
+  xlargeRadius,
+  mediumRadius,
+  largeRadius
 } from 'features/fireWeather/components/maps/stationAccuracy'
 
 describe('Station map color accuracy', () => {
-  const defaultMetrics = {
-    temperature: 0,
-    relative_humidity: 0
+  const nullObs: StationMetrics = {
+    observations: null,
+    forecasts: {
+      temperature: 15,
+      relative_humidity: 20
+    }
   }
-  describe('computeAccuracyColors', () => {
-    it('should return the neutral color code when observed and forecasted are equal', () => {
-      const exactForecast: StationMetrics = {
-        observations: defaultMetrics,
-        forecasts: defaultMetrics
+  const nullForecast: StationMetrics = {
+    forecasts: null,
+    observations: {
+      temperature: 20,
+      relative_humidity: 30
+    }
+  }
+  const perfectForecast: StationMetrics = {
+    observations: {
+      relative_humidity: 30,
+      temperature: 0
+    },
+    forecasts: {
+      relative_humidity: 30,
+      temperature: 0
+    }
+  }
+  describe('computeRHScaleIndex', () => {
+    it('should return the neutral index when observed and forecast RH are equal', () => {
+      expect(computeRHScaleIndex(100, 100)).toEqual(neutralIndex)
+    })
+
+    it('should return the largest scale index when observed RH is much lower than forecasted RH', () => {
+      expect(computeRHScaleIndex(100, 10)).toEqual(rhColorScale.length - 1)
+    })
+
+    it('should return a scale index of 0 when observed RH is much greater than forecasted RH', () => {
+      expect(computeRHScaleIndex(10, 100)).toEqual(0)
+    })
+
+    it('should return a scale index of neutralIndex+1 when observed RH is 4 percentage points lower than forecasted RH', () => {
```

Comment:
Minor: I thought it was 3 percentage points? Maybe this has changed or I'm remembering wrong.

Reply:
Each step is 3 percentage points. So that means if observed RH is 3 percentage points lower than forecasted RH, the forecast is considered "correct" and the neutral colour will be shown. This test is checking for what happens when the forecast is considered "slightly wrong", as in the map marker should not be a neutral colour.